### PR TITLE
Azure: Add EncryptionType field to ManagedDiskParameters

### PR DESCRIPTION
**Pull request by jhixson74**
This allows the type of Azure disk encryption to be specified.

https://issues.redhat.com/browse/CORS-1889
https://issues.redhat.com/browse/CORS-1890
https://issues.redhat.com/browse/CORS-1891

### CODE REVIEW

File: machine/v1beta1/types_azureprovider.go [hunk 1]
```diff
@@ -234,6 +234,10 @@ type ManagedDiskParameters struct {
 	// DiskEncryptionSet is the disk encryption set properties
 	// +optional
 	DiskEncryptionSet *DiskEncryptionSetParameters `json:"diskEncryptionSet,omitempty"`
+	// EncryptionType is the type of diisk encryption used.
+	// Possible values include: 'EncryptionAtRestWithPlatformKey', 'EncryptionAtRestWithCustomerKey', and 'EncryptionAtRestWithPlatformAndCustomerKeys'.
+	// +optional
+	EncryptionType string `json:"encryptionType,omitempty"`
```
Comment (JoelSpeed) [inline machine/v1beta1/types_azureprovider.go:240]: We tend to make enums like this a typed string with specific values as constants, the constants can then be used in implementation to validate the valid values

**Review by JoelSpeed — COMMENTED** [review]
Is there any enhancement or documentation that explains what this new field is doing? I don't know anything about these encryption types or how they might be used

Are there any caveats to these? Do customers need to set anything up before they can use the different types? Are there compatibility issues that mean other options are restricted when using these options?

File: machine/v1beta1/types_azureprovider.go (hunk 1 above)
Comment (m1kola) [reply machine/v1beta1/types_azureprovider.go:240]: As far as I know stuff in here must correspond to the Azure SDK/Client which in turn corresponds to Azure API. And there is no such filed in [the Azure API as encryption type](https://docs.microsoft.com/en-us/rest/api/compute/virtual-machines/create-or-update#manageddiskparameters).

**Review by m1kola — CHANGES_REQUESTED** [review]
@jhixson74 I believe no changes required in this repo to enable server side encryption or encryption at host. We already have `DiskEncryptionSet` and this is what we need.

**Pull request closed without merging**